File: js/sprint5/N/code.js
/**
Comment it before submitting 
class Node { 
    constructor(value, left = null, right = null, size = 0) { 
        this.value = value; 
        this.left = left; 
        this.right = right; 
        this.size = size;
    } 
}
**/

function split(node, k) {
    // Your code
    // “ヽ(´▽｀)ノ”
    return [null, null];
}

function test() {
    const node1 = new Node(3, null, null, 1);
    const node2 = new Node(2, null, node1, 2);
    const node3 = new Node(8, null, null, 1);
    const node4 = new Node(11, null, null, 1);
    const node5 = new Node(10, node3, node4, 3);
    const node6 = new Node(5, node2, node5, 6);
    const res = split(node6, 4);

    console.assert(res[0].size === 4);
    console.assert(res[1].size === 2);
}

File: go/sprint5/N/code.go
package main

/**
Comment it before submitting
type Node struct { 
	value    int 
	left   *Node 
	right  *Node 
	size     int
}
**/

func split(node *Node, k int) (*Node, *Node) {
    // Your code
    // “ヽ(´▽｀)ノ”
}

func test() {
	node1 := &Node{3, nil, nil, 1}
    node2 := &Node{2, nil, node1, 2}
    node3 := &Node{8, nil, nil, 1}
    node4 := &Node{11, nil, nil, 1}
    node5 := &Node{10, node3, node4, 3}
    node6 := &Node{5, node2, node5, 6}
    left, right := split(node6, 4);
    if left.size != 4 {
		panic("WA")
	}
	if right.size != 2 {
		panic("WA")
	}
}